src/ui/zcl_abapgit_services_git.clas.abap

```diff
@@ -93,6 +98,38 @@ ENDCLASS.
 
 CLASS zcl_abapgit_services_git IMPLEMENTATION.
 
+  METHOD checkout_commit.
+
+    DATA: lo_repo            TYPE REF TO zcl_abapgit_repo_online,
+          lt_value_tab       TYPE ty_commit_value_tab_tt,
+          lt_commits         TYPE zif_abapgit_definitions=>ty_commit_tt,
+          ls_selected_commit TYPE zif_abapgit_definitions=>ty_commit.
+
+    FIELD-SYMBOLS: <ls_field_desc> TYPE rsvbfidesc.
+
+    lo_repo ?= zcl_abapgit_repo_srv=>get_instance( )->get( iv_key ).
+
+    IF lo_repo->get_local_settings( )-write_protected = abap_true.
+      zcx_abapgit_exception=>raise( 'Cannot checkout. Local code is write-protected by repo config.' ).
+    ENDIF.
+
+    checkout_commit_build_list(
+      EXPORTING
+        iv_branch_name = lo_repo->get_branch_name( )
+        iv_url         = lo_repo->get_url( )
+      IMPORTING
+        et_value_tab   = lt_value_tab
+        et_commits     = lt_commits ).
+
+    ls_selected_commit = checkout_commit_build_popup( it_commits   = lt_commits
+                                                      it_value_tab = lt_value_tab ).
+
+    lo_repo->set_sha1( ls_selected_commit-sha1 ).
+    COMMIT WORK AND WAIT.
```

Comment:
How will the refreshing of the repository work? this just sets a sha1 value, ie. the repository might be in an inconsistent state?

Reply:
In set_sha1 calling reset_remote( ) was missing. Thanks 👍

Reply:
Maybe we should split this PR? It gets a little bit confusing, right?